src/wrapper/jnienv.rs

```diff
@@ -343,21 +341,28 @@ impl<'a> JNIEnv<'a> {
 
     /// Returns the starting address of the memory of the direct
     /// java.nio.ByteBuffer.
-    pub fn get_direct_buffer_address(&self, buf: JByteBuffer) -> Result<&mut [u8]> {
+    pub fn get_direct_buffer_address(&self, buf: JByteBuffer) -> Result<*mut u8> {
         non_null!(buf, "get_direct_buffer_address argument");
-        let ptr: *mut c_void =
-            jni_unchecked!(self.internal, GetDirectBufferAddress, buf.into_inner());
+        let ptr = jni_unchecked!(self.internal, GetDirectBufferAddress, buf.into_inner());
         non_null!(ptr, "get_direct_buffer_address return value");
-        let capacity = self.get_direct_buffer_capacity(buf)?;
-        unsafe { Ok(slice::from_raw_parts_mut(ptr as *mut u8, capacity as usize)) }
+        Ok(ptr.cast())
```

Comment:
What an expensive way to write `ptr as _`. ;-)

Reply:
yeah I guess I use `as _` and `cast()` arbitrarily, interchangeably - couldn't really say how come I used `cast` in this case

Reply:
ended up switching it when I made the other fix for the comment